include/fiction/algorithms/properties/check_planarity.hpp

```diff
@@ -0,0 +1,104 @@
+//
+// Created by benjamin on 17.07.24.
+//
+
+#ifndef FICTION_CHECK_PLANARITY_HPP
+#define FICTION_CHECK_PLANARITY_HPP
+
+#include "fiction/algorithms/network_transformation/network_balancing.hpp"
+
+#include <mockturtle/traits.hpp>
+
+#include <algorithm>
+#include <cstdint>
+#include <stdexcept>
+
+namespace fiction
+{
+
+template <typename Ntk>
+class check_planarity_impl
+{
+  public:
+    check_planarity_impl(const Ntk& ntk) : ntk(ntk) {}
+
+    /**
+     * @brief Checks if a given network is planar.
+     *
+     * This function checks if the network represented by the variable `ntk` is planar.
+     * The network is planar if for any edge with starting point m and endpoint n (represented by the node ranks), there
+     * is never another edge with starting point m_ > m and endpoint n_ < n, or vice versa. When iterating through
+     * the ranks of one level, the endpoints are always increasing. Therefore, only the starting points need to be
+     * checked. Thus, the highest connected starting point in the fan-in gives a border m_max for every subsequent edge.
```

Comment:
This description is really nice and could be added to the API function below, which is currently a bit sparsely documented. This docstring here is only visible to developers. The one below also to users.